### PR TITLE
improve countdown logic

### PR DESCRIPTION
**Pull request by ximex**
use `Intl.RelativeTimeFormat` to support different l10n
add week as possible duration

### CODE REVIEW

File: resources/assets/js/countdown.js [hunk 1]
```diff
@@ -1,81 +1,46 @@
...
+
+    return rtf.format(0, 'second');
+  }
+
   document.querySelectorAll('[data-countdown-ts]').forEach((element) => {
-    const timestamp = element.dataset.countdownTs;
-    const template = element.innerHTML;
-    element.innerHTML = template.replace('%c', formatFromNow(timestamp));
+    const timestamp = Number(element.dataset.countdownTs);
+    element.innerHTML = formatFromNow(timestamp);
```
Comment (weeman1337) [inline resources/assets/js/countdown.js:41]: Can you please roll back to keep the rest of the text and only replace `%c`? This works as kind of a template variable in the template.

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Thank you for your contribution @ximex 

Can you check my comment?

**Comment by weeman1337** [comment]
@MyIgel changes look good from my side though they change the countdown display slightly. What do you think?

Before

cd_before

After

cd_after

**Comment by ximex** [comment]
if the `übermorgen` etc naming is not wanted it's easy to remove by removing this option: `{ numeric: 'auto' }`. We only need to create another object with this option for `return rtf.format(0, 'second');` to get here `jetzt`.
The shift of one unit (minute, hours, etc): could check to change that the shown value is always smaller than the real time (real 3min20sec -> shown 3min or 4min)

File: resources/assets/js/countdown.js (hunk 1 above)
Comment (ximex) [reply resources/assets/js/countdown.js:41]: what do you mean? the template variable is completely removed. the complete text is generated by the browser API

**Comment by MyIgel** [comment]
I think having something like "tomorrow" is fine here but eventually we should also add a `title` attribute with the real datetime value

File: resources/assets/js/countdown.js (hunk 1 above)
Comment (weeman1337) [reply resources/assets/js/countdown.js:41]: Look for example at the design page https://github.com/engelsystem/engelsystem/blob/main/resources/views/pages/design.twig#L325-L338 There can be more text in the element. Just `%c` should be replaced with the countdown.

**Comment by weeman1337** [comment]
Picking up MyIgels comment: Can you add a title? I would recommend the full text „in 2 Tagen 23 Stunden und 42 Minuten“ (just an example; something like that; longest description you can get from the Browser API).

And if we are setting the title we can also add the date + time.

```
Nächste Schicht in 2 Tagen, 23 Stunden und 42 Minuten
13.12.2022 18:23
```

File: resources/assets/js/countdown.js (hunk 1 above)
Comment (ximex) [reply resources/assets/js/countdown.js]: now i got it. fixed

**Comment by ximex** [comment]
@weeman1337 thats not possible with this API

**Comment by MyIgel** [comment]
In that case just having a date and  time in the title attribute is enough imho

**Comment by ximex** [comment]
@MyIgel that way?

**Pull request merged**